state/applicationoffers_test.go

```diff
@@ -794,3 +794,40 @@ func (s *applicationOffersSuite) TestWatchOfferStatus(c *gc.C) {
 	wc.AssertOneChange()
 	wc.AssertNoChange()
 }
+
+func (s *applicationOffersSuite) TestWatchOffer(c *gc.C) {
+	ao := state.NewApplicationOffers(s.State)
+	offer, err := ao.AddOffer(crossmodel.AddApplicationOfferArgs{
+		OfferName:       "hosted-mysql",
+		ApplicationName: "mysql",
+		Owner:           s.Owner.Id(),
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	anotherOffer, err := ao.AddOffer(crossmodel.AddApplicationOfferArgs{
+		OfferName:       "hosted-postgresql",
+		ApplicationName: "mysql",
+		Owner:           s.Owner.Id(),
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	// Ensure that all the creation events have flowed through the system.
+	s.WaitForModelWatchersIdle(c, s.Model.UUID())
+
+	w := s.State.WatchOffer(offer.OfferName)
+
+	defer statetesting.AssertStop(c, w)
+	wc := statetesting.NewNotifyWatcherC(c, s.State, w)
+	// Initial event.
+	wc.AssertOneChange()
+	wc.AssertNoChange()
```

Comment:
All the AssertOneChange calls have an implicit AssertNoChange in them as well, so you could remove the extras here.